lightning/src/chain/channelmonitor.rs

```diff
@@ -1747,6 +1717,49 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 			};
 		let htlc_output_spend_pending = self.onchain_tx_handler.is_output_spend_pending(&htlc_output_to_spend);
 
+		for event in self.onchain_events_awaiting_threshold_conf.iter() {
+			match event.event {
+				OnchainEvent::HTLCUpdate { commitment_tx_output_idx, htlc_value_satoshis, .. }
+				if commitment_tx_output_idx == Some(htlc_commitment_tx_output_idx) => {
+					debug_assert!(htlc_spend_txid_opt.is_none());
+					htlc_spend_txid_opt = Some(&event.txid);
+					debug_assert!(htlc_spend_tx_opt.is_none());
+					htlc_spend_tx_opt = event.transaction.as_ref();
+					debug_assert!(holder_timeout_spend_pending.is_none());
+					debug_assert_eq!(htlc_value_satoshis.unwrap(), htlc.amount_msat / 1000);
+					holder_timeout_spend_pending = Some(event.confirmation_threshold());
+				},
+				OnchainEvent::HTLCSpendConfirmation { commitment_tx_output_idx, preimage, .. }
+				if commitment_tx_output_idx == htlc_commitment_tx_output_idx => {
+					debug_assert!(htlc_spend_txid_opt.is_none());
+					htlc_spend_txid_opt = Some(&event.txid);
+					debug_assert!(htlc_spend_tx_opt.is_none());
+					htlc_spend_tx_opt = event.transaction.as_ref();
+					debug_assert!(htlc_spend_pending.is_none());
+					htlc_spend_pending = Some((event.confirmation_threshold(), preimage.is_some()));
+				},
+				OnchainEvent::MaturingOutput {
+					descriptor: SpendableOutputDescriptor::DelayedPaymentOutput(_) }
+				if event.transaction.as_ref().map(|tx| tx.input.iter()
+					.any(|inp| Some(inp.previous_output.txid) == confirmed_txid &&
+						inp.previous_output.vout == htlc_commitment_tx_output_idx))
+					.unwrap_or(false)
+				=> {
+					debug_assert!(holder_delayed_output_pending.is_none());
+					holder_delayed_output_pending = Some(event.confirmation_threshold());
+				},
+				OnchainEvent::MaturingOutput {
+					descriptor: SpendableOutputDescriptor::DelayedPaymentOutput(ref descriptor) }
+				if descriptor.outpoint.into_bitcoin_outpoint() == htlc_output_to_spend => {
```

Comment:
IIUC this path should check we’re detecting well a second-stage HTLC transaction (HTLC-timeout or HTLC-preimage) CSV encumbered delayed and return further a `ClaimableAwaitingConfirmations` balance event